docs/topics/testing/advanced.txt

```diff
@@ -600,9 +600,13 @@ utility methods in the ``django.test.utils`` module.
 
 .. function:: setup_test_environment()
 
-    Performs any global pre-test setup, such as the installing the
-    instrumentation of the template rendering system and setting up
-    the dummy email outbox.
+    Performs global pre-test setup, like installing instrumentation for the
+    template rendering system and setting up the dummy email outbox.
+
+    This function also saves the values of settings that it changes so that
```

Comment:
I agree this is a bit obscure. On the other hand, if it ever does cause a problem, I could see it being subtle and hard to track down, etc. Raising an error occurred to me, but I was concerned about backwards incompatibility. If someone is calling it twice in real code, they will start getting errors where they weren't before. Would that be okay?

Another idea that just occurred to me: make `setup_test_environment()` a no-op if called more than once (which it is already capable of detecting).